File: app/src/main/java/igrek/songbook/service/system/WindowManagerService.java
package igrek.songbook.service.system;

import android.support.v7.app.AppCompatActivity;
import android.util.DisplayMetrics;
import android.view.WindowManager;

import javax.inject.Inject;

import igrek.songbook.dagger.DaggerIoc;

public class WindowManagerService {
	
	@Inject
	AppCompatActivity activity;
	
	public WindowManagerService() {
		DaggerIoc.getFactoryComponent().inject(this);
	}
	
	public void keepScreenOn(boolean set) {
		if (set) {
			activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
		} else {
			activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
		}
		setShowWhenLocked(set);
	}
	
	public void hideTaskbar() {
		if (activity.getSupportActionBar() != null) {
			activity.getSupportActionBar().hide();
		}
	}
	
	public void setFullscreen(boolean set) {
		int flag = WindowManager.LayoutParams.FLAG_FULLSCREEN;
		if (set) {
			activity.getWindow().addFlags(flag);
		} else {
			activity.getWindow().clearFlags(flag);
		}
	}
	
	public void setShowWhenLocked(boolean set) {
		int flag = WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON | WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED;
		if (set) {
			activity.getWindow().setFlags(flag, flag);
		} else {
			activity.getWindow().clearFlags(flag);
		}
	}
	
	public int getDpi() {
		DisplayMetrics metrics = activity.getResources().getDisplayMetrics();
		return metrics.densityDpi;
	}
	
	public float dp2px(float dp) {
		return dp * ((float) getDpi() / DisplayMetrics.DENSITY_DEFAULT);
	}
}

File: app/src/main/java/igrek/songbook/system/WindowManagerService.kt
package igrek.songbook.system


import android.util.DisplayMetrics
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import igrek.songbook.inject.LazyExtractor
import igrek.songbook.inject.LazyInject
import igrek.songbook.inject.appFactory

class WindowManagerService(
        appCompatActivity: LazyInject<AppCompatActivity> = appFactory.appCompatActivity,
) {
    private val activity by LazyExtractor(appCompatActivity)

    private val dpi: Int
        get() {
            val metrics = activity.resources.displayMetrics
            return metrics.densityDpi
        }

    fun keepScreenOn(set: Boolean) {
        if (set) {
            activity.window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        } else {
            activity.window.clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        }
        setShowWhenLocked(set)
    }

    fun hideTaskbar() {
        if (activity.supportActionBar != null) {
            activity.supportActionBar!!.hide()
        }
    }

    fun setFullscreen(set: Boolean) {
        val flag = WindowManager.LayoutParams.FLAG_FULLSCREEN
        if (set) {
            activity.window.addFlags(flag)
        } else {
            activity.window.clearFlags(flag)
        }
    }

    private fun setShowWhenLocked(set: Boolean) {
        val flag = WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON or WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
        if (set) {
            activity.window.setFlags(flag, flag)
        } else {
            activity.window.clearFlags(flag)
        }
    }

    fun dp2px(dp: Float): Float {
        return dp * (dpi.toFloat() / DisplayMetrics.DENSITY_DEFAULT)
    }

    fun showAppWhenLocked() {
        activity.window
                .addFlags(WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON or WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED)
    }
}
